server/datastore/mysql/software.go

```diff
@@ -961,14 +961,20 @@ func updateModifiedHostSoftwareDB(
 	var keysToUpdate []string
 	for key, newSw := range incomingMap {
 		curSw, ok := currentMap[key]
-		if !ok || newSw.LastOpenedAt == nil {
-			// software must also exist in current map, and new software must have a
-			// last opened at timestamp (otherwise we don't overwrite the old one)
-			if _, ok := existingBundleIDsToUpdate[newSw.BundleIdentifier]; !ok {
-				continue
+		// software must exist in current map for us to update it.
+		if !ok {
+			continue
+		}
```

Comment:
There were two things that could trigger a panic here:
1. if `ok` was false here, it means `curSw` was `nil`, so `curSw.LastOpenedAt` caused a panic
2. if `newSw.LastOpenedAt` was nil and `curSw.LastOpenedAt` wasn't, then `newSw.LastOpenedAt.Sub()` caused a panic.

The new logic continues immediately if `curSw` is nil.  If `newSw.LastOpenedAt` is `nil`, it checks that the item is marked as having a name change (by its presence in `existingBundleIDsToUpdate`), and also checks that the current item doesn't have a `LastOpenedAt` time.